gradoop-flink/src/main/java/org/gradoop/flink/io/impl/rdbms/metadata/MetaDataParser.java

```diff
@@ -0,0 +1,288 @@
+      ArrayList<NameTypeTuple> primaryKeys = new ArrayList<NameTypeTuple>();
+
+      // used to store foreign key metadata representation
+      ArrayList<FkTuple> foreignKeys = new ArrayList<FkTuple>();
+
+      // used to store further attributes metadata representation
+      ArrayList<NameTypeTuple> furtherAttributes = new ArrayList<NameTypeTuple>();
+
+      // used to find further attributes, respectively no primary or
+      // foreign key attributes
+      ArrayList<String> pkfkAttributes = new ArrayList<String>();
+
+      ResultSet rsPrimaryKeys = metadata.getPrimaryKeys(null, schemName, tableName);
+
+      // parses primary keys if exists
+      if (rsPrimaryKeys != null) {
+
+        // assigning primary key name
+        while (rsPrimaryKeys.next()) {
+          primaryKeys.add(new NameTypeTuple(rsPrimaryKeys.getString("COLUMN_NAME"), null));
+          pkfkAttributes.add(rsPrimaryKeys.getString("COLUMN_NAME"));
+        }
+
+        // assigning primary key data type
+        for (NameTypeTuple pk : primaryKeys) {
+          ResultSet rsColumns = metadata.getColumns(null, null, tableName, pk.f0);
+          rsColumns.next();
+          pk.f1 = JDBCType.valueOf(rsColumns.getInt("DATA_TYPE"));
+        }
+      }
+
+      ResultSet rsForeignKeys = metadata.getImportedKeys(null, schemName, tableName);
+
+      // parses foreign keys if exists
+      if (rsForeignKeys != null) {
+
+        // assigning foreign key name and name of belonging primary
+        // and foreign key table
+        while (rsForeignKeys.next()) {
+
+          String refdTableName = rsForeignKeys.getString("PKTABLE_NAME");
+          String refdTableSchem = rsForeignKeys.getString("PKTABLE_SCHEM");
+
+          if (refdTableSchem == null) {
+            foreignKeys.add(new FkTuple(rsForeignKeys.getString("FKCOLUMN_NAME"), null,
+                rsForeignKeys.getString("PKCOLUMN_NAME"), refdTableName));
+          } else {
+            foreignKeys.add(new FkTuple(rsForeignKeys.getString("FKCOLUMN_NAME"), null,
+                rsForeignKeys.getString("PKCOLUMN_NAME"), refdTableSchem + "." + refdTableName));
+          }
+
+          pkfkAttributes.add(rsForeignKeys.getString("FKCOLUMN_NAME"));
+        }
+
+        // assigning foreign key data type
+        for (FkTuple fk : foreignKeys) {
+          ResultSet rsColumns = metadata.getColumns(null, null, tableName, fk.f0);
+          rsColumns.next();
+          fk.f1 = JDBCType.valueOf(rsColumns.getInt("DATA_TYPE"));
+        }
+      }
+
+      ResultSet rsAttributes = metadata.getColumns(null, schemName, tableName, null);
+
+      // parses further attributes if exists
+      // assigning attribute name and belonging data type
+      while (rsAttributes.next()) {
+
+        // catches unsupported data types
+        if (!pkfkAttributes.contains(rsAttributes.getString("COLUMN_NAME")) &&
+            JDBCType.valueOf(rsAttributes.getInt("DATA_TYPE")) != JDBCType.OTHER &&
```

Comment:
Those String should be constants (especially since they are used multiple times)